src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp

```diff
@@ -16,7 +16,11 @@
 #include <string>
 #include <string.h>
 #include <sys/stat.h>
-#ifdef HAVE_SYS_SYSCTL_H
```

Comment:
The problem is that on NetBSD, there is no sys/sysclt.h header, but the sysctl is present. So the include has to be guarded by the HAVE_SYS_SYSCTL_H. See the commit https://github.com/josteink/coreclr/commit/0eafec3e1065ba68c10f065d2713aaea0f75c160